coralproject/talk
# Fix/corl 2327 user ban bug

Currently WIP:
- debugging an infinite update loop in UserStatusSitesList
- need to verify that API changes to BanModal don't break ModerateCardContainer, where it is also used
- need to update tests involving BanModal (having trouble getting `data-testid` to work on DropDownButton)

## What does this PR do?
This PR addresses https://vmproduct.atlassian.net/browse/CORL-1639 and https://vmproduct.atlassian.net/browse/CORL-2327.

- Site moderators can now unban commenters from sites within their moderation scope from the BanModal
- Commenters can be selectively unbanned from individual sites (rather than blanket unbanning them from all sites)

## What changes to the GraphQL/Database Schema does this PR introduce?
This adds the `updateUserBan` mutation, `UpdateUserBanInput` input and `UpdateUserBanPayload` type

## How do I test this PR?
1. Log in as an admin, navigate to the community page, and ban a user from several sites (preferably overlapping with some site moderator user's moderation scope). 
2. As an admin, unban that same user from one site, see that the user remains banned on the other sites
3. Log in as a site moderator, see that (for the commenter from above) all sites on which the user is banned show up in the checklist, with sites out of your scope disabled. Observe that you can unban that commenter from sites in your scope, as well as ban them from other sites in your scope.
 
## How do we deploy this PR?
No special steps should be required.
